src/components/actions/dev_tools/routes.ts

```diff
@@ -0,0 +1,264 @@
+import { actionLibrary } from '../../pluginsConnect'
+import { getApplicationData } from '.././getApplicationData'
+import { combineRequestParams } from '../../utilityFunctions'
+import DBConnect from '../../databaseConnect'
+import db from './databaseMethods'
+import { processTrigger } from '.././processTrigger'
+import {
+  ActionQueueStatus,
+  ApplicationOutcome,
+  ApplicationStatus,
+  Decision,
+  Trigger,
+} from '../../../generated/graphql'
+import {
+  selectRandomReviewAssignment,
+  getRandomReviewId,
+  getApplicationBasics,
+  mapTriggerShortcut,
+} from './helpers'
+import { ActionApplicationData, ActionResult, TriggerPayload } from '../../../types'
+
+// These routes should only be used for testing in development. They should
+// NEVER be used in the app.
+
+export const routeRunAction = async (request: any, reply: any) => {
+  const { actionCode, applicationId, reviewId, parameters } = combineRequestParams(request, 'camel')
+  const applicationData = applicationId ? await getApplicationData({ applicationId, reviewId }) : {}
+  const actionResult = await actionLibrary[actionCode]({
+    parameters,
+    applicationData,
+    DBConnect,
+  })
+  return reply.send(actionResult)
+}
+
+interface RequestProps {
+  templateCode: string
+  trigger: Trigger | 'RESET'
+  assignmentId?: number
+  sections?: string[]
+  reviewId?: number
+  decision?: Decision
+  comment?: string
+  eventCode?: string
+  applicationId?: number
+  serial?: string
+  stageNumber?: number
+  status?: ApplicationStatus
+  outcome?: ApplicationOutcome
+  applicationDataOverride: Partial<ActionApplicationData>
+}
+
+export const routeTestTrigger = async (request: any, reply: any) => {
+  const params: RequestProps = combineRequestParams(request, 'camel')
+  let { applicationId, serial } = params
+  const {
+    templateCode,
+    trigger,
+    assignmentId,
+    sections,
+    reviewId,
+    decision,
+    comment,
+    eventCode,
+    stageNumber,
+    status,
+    outcome,
+    applicationDataOverride = {},
+  }: RequestProps = params
+
+  const triggerFull = mapTriggerShortcut(trigger)
+
+  const { configId, configSerial, templateId, sectionCodes } = await db.getConfigApplicationInfo(
+    templateCode
+  )
+
+  if (!configId) reply.send('Invalid template code, or no config application available')
```

Comment:
Didn't understand: this would run even if the user defined a different applicationId? Is this to check if the template and config application is defined to avoid other further problems?

Reply:
Yes, you can run it for any application you like. By default it'll use the config application (cos that's easier and it always exists), but you can test it for a specific application in a specific state if you want. :)

Reply:
I understood that you have the 2 options. What I was asking was: `this would run even if the user defined a different applicationId?` not really sure why it's checking `configId` even if the user passes another applicationId to use.